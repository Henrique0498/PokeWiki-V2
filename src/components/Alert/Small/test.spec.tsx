import {
  getByTitle,
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved
} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import AlertSmall, { AlertSmallProps } from '.'

// describe('<SmallAlert />', () => {
//   it('Should render the alert small', () => {
//     const { container } = render(
//       <SmallAlert children="react avançado" type="info" />
//     )

//     expect(
//       screen.getByRole('heading', { name: /react avançado/i })
//     ).toBeInTheDocument()

//     expect(container.firstChild).toMatchSnapshot()
//   })

//   it('should render the colors correctly', () => {
//     const { container } = render(<SmallAlert />)

//     expect(container.firstChild).toHaveClass(
//       'bg-gray-300 dark:bg-gray-600 text-gray-700 dark:text-gray-100'
//     )
//   })
// })
let value = 'testing'

const mock: AlertSmallProps = {
  children: value,
  type: 'info',
  setMessage: () => {
    value = ''
  }
}

// test('sum', () => {
//   const { getByText } = render(<SmallAlert {...mock} />)

//   expect(getByText('testing')).toBeTruthy()
// })

describe('<SmallAlert/>', () => {
  it('should render component', () => {
    const { getByText } = render(<AlertSmall {...mock} />)

    expect(getByText('testing')).toBeInTheDocument()
  })

  it('should be able to add new item to list', async () => {
    const { queryByText, getByTitle } = render(<AlertSmall {...mock} />)

    const closeAlert = getByTitle('close')

    userEvent.click(closeAlert)

    await waitFor(() => {
      expect(queryByText('testing')).toBeInTheDocument()
    })
  })

  it('should render correctly', () => {
    const { container } = render(<AlertSmall {...mock} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should color correctly', () => {
    const { container } = render(<AlertSmall {...mock} />)

    expect(container.firstChild).toHaveStyle({ 'border-radius': '0.125rem' })
  })
})
