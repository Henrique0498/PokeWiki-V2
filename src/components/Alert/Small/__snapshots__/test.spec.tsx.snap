// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmallAlert/> should render correctly 1`] = `
<div
  class="flex p-2 w-full rounded-sm
        bg-gray-300 dark:bg-gray-600 text-gray-700 dark:text-gray-100 undefined
      "
>
  <p
    class="w-full"
  >
    testing
  </p>
  <button
    title="close"
    type="button"
  >
    <svg
      class="h-4 w-4 "
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
